lib/httpResponse.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { HttpRequest } from "./httpRequest";
+import { HttpHeaders } from "./httpHeaders";
+
+/**
+ * A HTTP response that is received from a server after a HttpRequest has been sent.
+ */
+export interface HttpResponse {
+    /**
+     * The request that was the cause of this HttpResponse.
+     */
+    request: HttpRequest;
+
+    /**
+     * The status code number of this response.
+     */
+    statusCode: number;
+
+    /**
+     * The HTTP headers of this response.
+     */
+    headers: HttpHeaders;
+
+    /**
+     * Get the body of this HttpResponse as a string.
+     */
+    bodyAsText(): Promise<string | undefined>;
+
+    /**
+     * Get the body of this HttpResponse as a JSON object.
+     */
+    bodyAsJson(): Promise<{} | any[] | undefined>;
```

Comment:
This is the return value of JSON.parse or xml2js.parseString, right? Let's call it parsedBody and IMO to avoid overengineering it just let it be a Promise<any> (since otherwise describing the types of values contained in it requires a recursive type definition that I can't think of a name for)

Reply:
I'm trying to keep the `bodyAs...()` methods similarly named, so how about we called them `textBody()` and `deserializedBody()`. We don't use the term "parsing" anywhere in our RequestPolicies, but we do we the terms "serializing" and "deserializing", so I feel like "deserialized" would be more familiar and would fit better.

Reply:
Now I remember that this is about "deserializing in the pipeline". So I understand that this isn't the return value of a JSON/XML.parse method, it's actually the return value of serializer.deserialize. deserializedBody seems fine.